src/invenio-records-js/controllers/InvenioRecordsCtrl.js

```diff
@@ -155,78 +155,83 @@ function InvenioRecordsCtrl($scope, $rootScope, $q, $window, $location,
   }
 
   /**
-    * Records actions
+    * Do a data massage before sending with request
     * @memberof InvenioRecordsCtrl
-    * @function invenioRecordsActions
-    * @param {Object} evt - The event object.
-    * @param {String} type - The invenio action type.
-    * @param {String} method - The invenio request method.
-    * @param {Object} successCallback - Call function after success.
-    * @param {Object} errorCallback - Call function after error..
+    * @function cleanData
     */
-  function invenioRecordsActions(evt, type, method, successCallback, errorCallback) {
+  function cleanData() {
+    var _data = angular.merge({}, {metadata: vm.invenioRecordsModel});
```

Comment:
Wrapping the model in the `metadata` key here, prevents the data from being cleaned below by the loop. So we should probably either pass the data through the cleaning loop first and wrap it afterwards in the `metadata`, or do `angular.forEach(_data.metadata, ...) { ... }`